### PR TITLE
Testing tools reedit

### PR DESCRIPTION
**Pull request by spence-man**
- Tested from scratch using README install guide
- Acceptance, Functional, & Unit tests run on separate port

[Staging site](https://chappress-staging.chapman.edu/)
[README](https://github.com/chapmanu/chap-press/tree/testing-tools-reedit#run-automated-tests)
[Trello Story](https://trello.com/c/SmvcHE6y/18-as-a-developer-i-want-installation-and-testing-simplified-for-new-developers)
[Wiki](https://github.com/chapmanu/chap-press/wiki/Automated-Testing)

### CODE REVIEW

File: config/templates/wp-config.php.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@
```
Comment (tatwell) [inline config/templates/wp-config.php.erb:15]: Is this template used? I'm not clear of difference between this and `dist` version below.

File: content/set-environment.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (tatwell) [inline content/set-environment.php]: Given how short and direct this is, can we include this directly in `wp-config`?

File: content/themes/chappress/main.scss [hunk 3]
```diff
@@ -15,49 +15,32 @@
...
 */
 
-@import "./sass/_smc_styles";
-@import "./sass/_entry";
-@import "./sass/_headers";
-@import "./sass/_top-menu";
-@import "./sass/_sidebar";
-@import "./sass/_login";
+@import "./partials/_smc_styles";
+@import "./partials/_login";
```
Comment (tatwell) [inline content/themes/chappress/main.scss:19]: I like the simplification here.

**Review by tatwell — COMMENTED** [review]
Looks good. I still need to test the build from scratch. Will do that tomorrow morning.

**Comment by tatwell** [comment]
I'm trying to install from scratch. On first step, I hit this:

```
$ git clone git@github.com:chapmanu/chap-press.git
Cloning into 'chap-press'...
The authenticity of host 'github.com (192.30.255.113)' can't be established.
RSA key fingerprint is SHA256:...
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'github.com,192.30.255.113' (RSA) to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I didn't have this problem last time I tried. Have repo permissions changed?

I'll try proceeding with the read-only HTTP URL.

---

#### Update

Yes, this was an issue on my laptop. The ssh key had got wiped out. Issue #17 got me sorted out.

File: config/templates/wp-config.php.erb (hunk 1 above)
Comment (spence-man) [reply config/templates/wp-config.php.erb:15]: Yes, used during Capistrano deployment (`deploy.rb`). Differences: Composer autoload path, pulls DB info from `config/database.yml`

**Comment by spence-man** [comment]
@tatwell No repo permissions changed as far as I know. Not sure if this [troubleshoot](https://github.com/chapmanu/chap-press/issues/17) will help. Found this [article](https://help.github.com/articles/error-permission-denied-publickey/) as well.

**Comment by tatwell** [comment]
@spence-man Looking at my bash history, it appears I was able to use the git url previously. Odd. But, yes, probably an issue with my machine.

I cloned SSH verison. At `wp-cli` steps, I'm hitting this error:

```
$ wp core install --url=http://localhost:8222/ --title=chap-press --admin_user=chappress --admin_password=password --admin_email=chappress@gmail.com
PHP Fatal error:  Call to undefined function do_action() in /usr/local/Cellar/wp-cli/1.4.0/php/WP_CLI/Runner.php(1070) : eval()'d code on line 90
```

---

#### Update

I think the issue here was that mysql/mariadb wasn't running. Once I got that sorted out, it went away.

**Comment by tatwell** [comment]
Unit and functional tests pass! And all the acceptance tests pass! ... except the last one:

```
$ codecept run acceptance
Codeception PHP Testing Framework v2.3.6
Powered by PHPUnit 5.7.23 by Sebastian Bergmann and contributors.

Acceptance Tests (4) --------------------------------------------------------------------------------------------------------------------------------

  [RunProcess] Starting php -S 127.0.0.1:8777 -t ./public
✔ expectToActivatePluginCept: Log in as an admin and activate plugin (5.92s)
✔ expectUserAndAdminLoginCest: Log in as a user (3.93s)
✔ expectUserAndAdminLoginCest: Log in as an admin (4.62s)
E expectUserAndAdminLoginCest: Login as an admin and add a new user (4.77s)
-----------------------------------------------------------------------------------------------------------------------------------------------------
  [RunProcess] Stopping php -S 127.0.0.1:8777 -t ./public


Time: 24.74 seconds, Memory: 18.25MB

There was 1 error:

---------
1) expectUserAndAdminLoginCest: Login as an admin and add a new user
 Test  tests/acceptance/expectUserAndAdminLoginCest.php:expectsAddNewUser
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
  [Facebook\WebDriver\Exception\UnknownServerException] unknown error: Element is not clickable at point (237, 638)
  (Session info: chrome=61.0.3163.100)
  (Driver info: chromedriver=2.30.477690 (c53f4ad87510ee97b5c3425a14c0e79780cdf262),platform=Mac OS X 10.12.5 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 43 milliseconds
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'SMC-MBOOKP13-01', ip: 'fe80:0:0:0:1c30:47ba:44e1:aa4c%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.5', java.version: '1.8.0_144'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, networkConnectionEnabled=false, chrome={chromedriverVersion=2.30.477690 (c53f4ad87510ee97b5c3425a14c0e79780cdf262), userDataDir=/var/folders/8y/d69h1zss11n3l8x___5gsnbx3sms60/T/.org.chromium.Chromium.nucGsF}, takesHeapSnapshot=true, pageLoadStrategy=normal, databaseEnabled=false, handlesAlerts=true, hasTouchScreen=false, version=61.0.3163.100, platform=MAC, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true, unexpectedAlertBehaviour=}]
Session ID: c565500028f1e98861eaf144a3eb7bb4  
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        

Scenario Steps:

 8. $I->click("Add New User") at tests/acceptance/expectUserAndAdminLoginCest.php:40
 7. // I am going to submit form and add new user
 6. $I->fillField("Email","example@example.com") at tests/acceptance/expectUserAndAdminLoginCest.php:38
 5. $I->fillField("Username","chappress-user") at tests/acceptance/expectUserAndAdminLoginCest.php:37
 4. $I->amOnPage("/wp-admin/user-new.php") at tests/acceptance/expectUserAndAdminLoginCest.php:36
 3. $I->loginAsAdmin() at tests/acceptance/expectUserAndAdminLoginCest.php:35

#1  /Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Exception/WebDriverException.php:114
#2  /Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/HttpCommandExecutor.php:320
#3  /Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php:535
#4  /Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/RemoteExecuteMethod.php:40
#5  /Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/RemoteWebElement.php:82
#6  Codeception\Module\WebDriver->click
#7  /Users/atwell/projects/chap-press/tests/_support/_generated/AcceptanceTesterActions.php:3061
#8  /Users/atwell/projects/chap-press/tests/acceptance/expectUserAndAdminLoginCest.php:40
#9  expectUserAndAdminLoginCest->expectsAddNewUser

ERRORS!
Tests: 4, Assertions: 3, Errors: 1.
```

Is it a possible timing issue? I know sometimes you have to wait for page rendering on js tests. 

I ran it several times and got same result each time.

When I watch the browser, it looks like it's on the right page and should be able to add user. Not sure why it fails.

**Review by tatwell — COMMENTED** [review]
One acceptance test still failing. I think that's the only thing standing between this PR and me approving it.

**Comment by spence-man** [comment]
@tatwell Ok, I'll look into that last test. Can you post the code when you run it as `codecept run acceptance -vvv `?

**Comment by tatwell** [comment]
@spence-man Verbose output from failed test:

```
expectUserAndAdminLoginCest: Login as an Admin and add a new user
Signature: expectUserAndAdminLoginCest:expectsAddNewUser
Test: tests/acceptance/expectUserAndAdminLoginCest.php:expectsAddNewUser
Scenario --
  Dump file domain [http://localhost:8222] replaced with [http://localhost:8777]
 I am on page "/"
  [GET] http://localhost:8777/
 I am going to log in as Admin
 I login as admin 
  [GET] http://localhost:8777/wp-login.php
  Cookies matching pattern /^wordpress_[a-z0-9]{32}$/ : {"wordpress_fe9e30d98e702e2c7d71b7af6ff322cc": "chappress%7C1508705133%7C1K6otZFHmdqoTzBPqXHE9dFA8kGdfOdwdmZgVqKIXyv%7C84eadd3c7f0509e2898e0661317c312cfeccbb0326f6db84544d2e0b4253cde3"}
  Cookies matching pattern /^wordpress_logged_in_[a-z0-9]{32}$/ : {"wordpress_logged_in_fe9e30d98e702e2c7d71b7af6ff322cc": "chappress%7C1508705133%7C1K6otZFHmdqoTzBPqXHE9dFA8kGdfOdwdmZgVqKIXyv%7C7fc763217f02b96ff1448d080c068ebd0f691311533006d22916d015a698d405"}
 I am on page "/wp-admin/user-new.php"
  [GET] http://localhost:8777/wp-admin/user-new.php
 I fill field "Username","chappress-user"
 I fill field "Email","example@example.com"
 I am going to submit form and add new user
 I click "Add New User"
  Unable to retrieve Selenium logs : date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone.
  Screenshot and page source were saved into '/Users/atwell/projects/chap-press/tests/_output/' dir
 ERROR 

-----------------------------------------------------------------------------------------------------------------------------------------------------
  [RunProcess] Stopping php -S 127.0.0.1:8777 -t ./public


Time: 22.77 seconds, Memory: 20.50MB

There was 1 error:

---------
1) expectUserAndAdminLoginCest: Login as an admin and add a new user
 Test  tests/acceptance/expectUserAndAdminLoginCest.php:expectsAddNewUser
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
  [Facebook\WebDriver\Exception\UnknownServerException] unknown error: Element is not clickable at point (237, 638)
  (Session info: chrome=61.0.3163.100)
  (Driver info: chromedriver=2.30.477690 (c53f4ad87510ee97b5c3425a14c0e79780cdf262),platform=Mac OS X 10.12.5 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 49 milliseconds
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'SMC-MBOOKP13-01', ip: 'fe80:0:0:0:1c30:47ba:44e1:aa4c%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.5', java.version: '1.8.0_144'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, networkConnectionEnabled=false, chrome={chromedriverVersion=2.30.477690 (c53f4ad87510ee97b5c3425a14c0e79780cdf262), userDataDir=/var/folders/8y/d69h1zss11n3l8x___5gsnbx3sms60/T/.org.chromium.Chromium.0Q7vbq}, takesHeapSnapshot=true, pageLoadStrategy=normal, databaseEnabled=false, handlesAlerts=true, hasTouchScreen=false, version=61.0.3163.100, platform=MAC, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true, unexpectedAlertBehaviour=}]
Session ID: fe1180211fa64f4726a0e89f564fdc45  
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        

Scenario Steps:

 8. $I->click("Add New User") at tests/acceptance/expectUserAndAdminLoginCest.php:40
 7. // I am going to submit form and add new user
 6. $I->fillField("Email","example@example.com") at tests/acceptance/expectUserAndAdminLoginCest.php:38
 5. $I->fillField("Username","chappress-user") at tests/acceptance/expectUserAndAdminLoginCest.php:37
 4. $I->amOnPage("/wp-admin/user-new.php") at tests/acceptance/expectUserAndAdminLoginCest.php:36
 3. $I->loginAsAdmin() at tests/acceptance/expectUserAndAdminLoginCest.php:35

/Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Exception/WebDriverException.php:114
/Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/HttpCommandExecutor.php:320
/Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php:535
/Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/RemoteExecuteMethod.php:40
/Users/atwell/projects/chap-press/vendor/facebook/webdriver/lib/Remote/RemoteWebElement.php:82
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Module/WebDriver.php:955
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Step.php:260
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Scenario.php:72
/Users/atwell/projects/chap-press/tests/_support/_generated/AcceptanceTesterActions.php:3061
/Users/atwell/projects/chap-press/tests/acceptance/expectUserAndAdminLoginCest.php:40
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Lib/Di.php:127
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Test/Cest.php:136
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Test/Cest.php:148
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Test/Cest.php:82
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Test/Test.php:87
/Users/atwell/projects/chap-press/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/PHPUnit/Runner.php:106
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/SuiteManager.php:157
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Codecept.php:188
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Codecept.php:157
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Command/Run.php:420
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Command/Run.php:347
/Users/atwell/projects/chap-press/vendor/symfony/console/Command/Command.php:262
/Users/atwell/projects/chap-press/vendor/symfony/console/Application.php:888
/Users/atwell/projects/chap-press/vendor/symfony/console/Application.php:224
/Users/atwell/projects/chap-press/vendor/symfony/console/Application.php:125
/Users/atwell/projects/chap-press/vendor/codeception/codeception/src/Codeception/Application.php:103
/Users/atwell/projects/chap-press/vendor/codeception/codeception/codecept:36
```

**Comment by spence-man** [comment]
@tatwell Updated `acceptance.suite.yml` to try to solve _Login as an admin and add a new user_ error. Changed from `wait:2` to `wait:5`

Wait (default: 0 seconds) - Whenever element is required and is not on page, wait for n seconds to find it before fail.
[WebDriver](http://codeception.com/docs/modules/WebDriver)

Otherwise, not sure what the error is.

**Comment by tatwell** [comment]
@spence-man I thought that might work but I'm still seeing an error.

I'll open a defect on it so we can merge this and move forward.

When you finish the Ansible story, you can work on that defect.

**Review by tatwell — APPROVED** [review]

**Pull request merged**